pkg/net/libp2p/libp2ptransport.go

```diff
@@ -173,37 +176,44 @@ func (t *Transport) connectToNode(ctx context.Context, addr multiaddr.Multiaddr)
 
 	s, err := t.openStream(ctx, info.ID)
 	if err != nil {
-		t.logger.Log(logging.LevelError, fmt.Sprintf("couldn't open stream: %v", err))
-		return nil, fmt.Errorf("couldn't open stream to %v: %w", addr, err)
+		return nil, fmt.Errorf("failed to open new stream to node %v: %w", addr, err)
 	}
 
 	return s, nil
 }
 
 func (t *Transport) openStream(ctx context.Context, p peer.ID) (network.Stream, error) {
-	for {
-		sctx, cancel := context.WithTimeout(ctx, defaultMaxTimeout)
-		s, err := t.host.NewStream(sctx, p, ID)
+	var streamErr error
+	for i := 0; i < retryAttempts; i++ {
+		sctx, cancel := context.WithTimeout(ctx, maxConnectingTimeout)
+
+		s, streamErr := t.host.NewStream(sctx, p, ProtocolID)
 		cancel()
```

Comment:
Do we need to cancel the context here?
If not, it's a bit confusing to still cancel it.

Reply:
Yes we do.

```
WithTimeout returns WithDeadline(parent, time.Now().Add(timeout)).

Canceling this context releases resources associated with it,
so code should call cancel as soon as the operations running in this Context complete
```

Reply:
Ok I said nothing :)